Python/ceval.c

```diff
@@ -2124,6 +2124,19 @@ _PyEval_UnpackIterableStackRef(PyThreadState *tstate, _PyStackRef v_stackref,
         _PyErr_Format(tstate, PyExc_ValueError,
                       "too many values to unpack (expected %d)",
                       argcnt);
+        PyObject *exc = _PyErr_GetRaisedException(tstate);
+        if (PyObject_SetAttr(exc, &_Py_ID(_unpacked_value), v)) {
+            _PyErr_ChainExceptions1(exc);
+            goto Error;
+        }
+        if (PyObject_SetAttr(exc,
+                             &_Py_ID(_unpack_expected_argcnt),
+                             PyLong_FromLong(argcnt))) {
```

Comment:
The `PyLong_FromLong` call could return NULL. You need to check it separately.

Reply:
Not only that, this is leaking the result

Reply:
as in PyLong_FromLong allocates?

Reply:
Got it, so now if unpack exception becomes a new exception subclass, the freeing logic can be made part of it probably?

Reply:
I'll add a test with a larger generator.